### DISCUSSION TITLE
landscape mountain segmentation

### DISCUSSION THREAD

**Discussion by keep9oing** [microsoft/AirSim#3512] [Support Q&A]
Hi, I'm studying using airsim and environments

I want to segment the road in the mountain. such as below

![airsim_q_a](https://user-images.githubusercontent.com/31655488/93419536-0b113680-f8e8-11ea-8211-ab85e7d2f26d.png)

However, the segmentation is done like landscape terrain and road is the same object. How can I separate them in the scene?

**Comment by ykhedar** [comment]
I think you are only able to see objects in the segmentation view which are static meshes, like the trees, bridges, stones etc.. the dirt road is actually part of the terrain(landscape) itself. So you would not see it separate from the landscape terrain.

**Comment by kharyal** [comment]
While trying to run AirSim on environments from the Unreal Marketplace, I am facing the following issue in getting correct segmentation images
![28_scene](https://user-images.githubusercontent.com/33199820/98181909-72cc2100-1f2a-11eb-9b94-d4550af111e7.png)
![28_seg](https://user-images.githubusercontent.com/33199820/98181905-719af400-1f2a-11eb-989d-e9bcc6edc6c7.png)

I want separate segmentation for road and believe that the above issue is happening because the meshes are stored in a Blueprint Class Actor. Is there any way to penetrate through the BP to get correct segmentation?

TIA

**Comment by kcaisfather** [comment]
Is there any solution for this inseparable segmentation issue. I have the same problem. We want to train our model to find drivable area with semantic segmentation method. Is there any opinion to deal with this in UE platform?

**Comment by kharyal** [comment]
I did that by porting the whole environment to Unity, separating individual meshes and then importing it back to Unreal engine. But, it has its own set of problems like textures might not appear in Unity.

-Regards

**Reply by dellhuyongcai** [reply]
Have you solved your problem?Did any other useful method that you use in your getting correct segmentation images,very grateful to get your replay.

**Reply by dellhuyongcai** [reply]
Have you solved your problem?Did any other useful method that you use in your getting correct segmentation images,very grateful to get your reply.

**Reply by kharyal** [reply]
The only way I could find was to separate individual meshes from the object. At that time, there was no way to do that in Unreal, therefore, I had to port the meshes to unity, separate the meshes, export them, and import them back into unreal. I am not aware if there's another way to do this directly in Unreal now (maybe check on [Unreal engine forums](https://forums.unrealengine.com/)?)

**Reply by dellhuyongcai** [reply]
All right,thanks a lot,I did it by adjust the road and the lane,generated the road firstly,then added the lane to the road secondly,make sure the road and lane are two different meshes,but it wasted my much time,thanks again.
